src/backend/executor/CfgBuilder.js

```diff
@@ -0,0 +1,34 @@
+define(["./InstructionSet"], function(InstructionSet) {
+
+    // cache to avoid cycles in loops
+    var cache = {};
+
+    function build(cfgTemplate) {
+        cache = {};
+        return buildNode(cfgTemplate, 0);
+    }
+
+    function buildNode(cfgTemplate, nodeId) {
+        if (cache[nodeId]) {
+            return cache[nodeId];
+        }
+
+        var templItem = cfgTemplate[nodeId];
+        if (!templItem) {
+            return;
+        }
+        var Instruction = InstructionSet[templItem.type];
+        if (!Instruction) {
+            throw new Error("Unknown VM instruction: " + templItem.type);
+        }
+        var node = new Instruction(templItem.param);
+        cache[nodeId] = node;
+        node.next = buildNode(cfgTemplate, templItem.next || (Number(nodeId) + 1));
+        // for BRANCH instruction
```

Comment:
If it, instead of filling true and false with undefined. Will make dealing with errors in the BRANCH instruction easier.